### DISCUSSION TITLE
How to disable the config.log?

### DISCUSSION THREAD

**Discussion by chenrui333** [sparklemotion/sqlite3-ruby#394] [Q&A]
👋 while trying to upgrade mailcatcher to [0.9.0 release](https://github.com/Homebrew/homebrew-core/pull/137980), I ran into the following failure

```
      The offending files are:
        libexec/gems/sqlite3-1.6.3/ext/sqlite3/tmp/arm64-apple-darwin20.6.0/ports/sqlite3/3.42.0/sqlite-autoconf-3420000/config.log

```

The file was generated from the sqlite3, I think. But I still got that generated even with the build options with as below:

```
==> gem install /Users/rui/Library/Caches/Homebrew/downloads/0048667fe771138b86aacef26b55defc9144ff7d23d17027cbc83c99c3a0df67--sqlite3-1.6.3.gem --ignore-dependencies --no-document --install-dir /opt/homebrew/Cellar/mailcatcher/0.9.0/libexec -- --use-system-libraries --with-sqlite3-include=/opt/homebrew/opt/sqlite/include --with-sqlite3-lib=/opt/homebrew/opt/sqlite/lib --with-cflags=-fdeclspec
Building native extensions with: '--use-system-libraries --with-sqlite3-include=/opt/homebrew/opt/sqlite/include --with-sqlite3-lib=/opt/homebrew/opt/sqlite/lib --with-cflags=-fdeclspec'
This could take a while...
Successfully installed sqlite3-1.6.3
```

nb adding `--platform=ruby` does not help either.

Let me how I can proceed it. Thanks!

**Comment by flavorjones** [comment]
Can you help me understand why the presence of a file named `config.log` is an issue?

Please provide more context on the problem you're seeing. What command did you run, what is the full output, etc.

I also noticed that the arguments to your `gem install` command are strange to me. Why are you installing into homebrew directories? Why are you installing sqlite3 into the mailcatcher directory?

You may want to try to use a precompiled version of sqlite3, if your platform is supported, which would sitestep the compile-upon-installation process that's generating the config.log file.

**Comment by iMichka** [comment]
From my understanding, config.log contains some references to what we call "compiler shims": we don't call call clang/gcc directly, we use our internal scripts to add some logs / modify compiler flags in a controlled manner.
We have an audit that check if files we ship to our users contain references to these "shims", as we don't want that to happen: these shims are unusable outside of Homebrew, and we have some mechanisms in place to replace the shims paths with real compiler paths.

So we do not want to ship these paths. One solution I see is to delete this "config.log".

> Why are you installing sqlite3 into the mailcatcher directory?

We call this "vendoring". We have 2 solutions for dependencies: a homebrew package (formula), or to ship dependencies into the package right away (vendoring): this second solution is often used for Ruby and Python dependencies, as we do not have Homebrew packages (formula) for each dependency out there.

> You may want to try to use a precompiled version of sqlite3

We don't do that: each package / dependency has to be compiled from source: our users trust us to build binaries ourselves and not to download precompiled versions from somewhere else (even if these precompiled versions are legit).

--- 

I think the concern is that `gem install` will install `config.log`: is that a file that is usually shipped to end users?
